tests/test_web_sendfile_functional.py

```diff
@@ -238,47 +243,62 @@ async def handler(request):
     app.router.add_get("/", handler)
     client = await aiohttp_client(app)
 
-    resp = await client.get("/")
+    resp = await client.get("/", headers={"Accept-Encoding": accept_encoding})
     assert resp.status == 200
-    body = await resp.read()
-    assert b"hello aiohttp\n" == body
-    assert resp.headers["Content-Type"] == "application/pdf"
-    assert resp.headers.get("Content-Encoding") == "gzip"
+    assert (
+        resp.headers["Content-Type"],
+        resp.headers.get("Content-Encoding"),
+        await resp.read(),
+    ) == ("application/pdf", expect_encoding, b"hello aiohttp\n")
     resp.close()
     await resp.release()
     await client.close()
 
 
-async def test_static_file_with_gziped_counter_part_enable_compression(
-    aiohttp_client: Any, sender: Any
+@pytest.mark.parametrize(
+    ("accept_encoding", "expect_encoding"),
+    [("gzip, deflate", "gzip"), ("gzip, deflate, br", "br")],
+)
+@pytest.mark.parametrize("forced_compression", [None, web.ContentCoding.gzip])
+async def test_static_file_with_encoding_and_enable_compression(
+    aiohttp_client: Any,
+    sender: Any,
+    accept_encoding: str,
+    expect_encoding: str,
+    forced_compression: Optional[web.ContentCoding],
 ):
-    """Test that enable_compression does not double compress when a .gz file is also present."""
+    """Test that enable_compression does not double compress when a static compressed file is also present."""
     filepath = pathlib.Path(__file__).parent / "hello.txt"
 
     async def handler(request):
         resp = sender(filepath)
-        resp.enable_compression()
+        resp.enable_compression(forced_compression)
         return resp
 
     app = web.Application()
     app.router.add_get("/", handler)
     client = await aiohttp_client(app)
 
-    resp = await client.get("/")
+    resp = await client.get("/", headers={"Accept-Encoding": accept_encoding})
     assert resp.status == 200
-    body = await resp.read()
-    assert body == b"hello aiohttp\n"
-    assert resp.headers["Content-Type"] == "text/plain"
-    assert resp.headers.get("Content-Encoding") == "gzip"
+    assert (
+        resp.headers["Content-Type"],
+        resp.headers.get("Content-Encoding"),
+        await resp.read(),
+    ) == ("text/plain", expect_encoding, b"hello aiohttp\n")
     resp.close()
     await resp.release()
     await client.close()
 
 
+@pytest.mark.parametrize(
+    ("extension", "expect_encoding"), [(".gz", "gzip"), (".br", "br")]
+)
 async def test_static_file_with_content_encoding(
-    aiohttp_client: Any, sender: Any
+    aiohttp_client: Any, sender: Any, extension: str, expect_encoding: str
 ) -> None:
-    filepath = pathlib.Path(__file__).parent / "hello.txt.gz"
+    """Test requesting of static compressed files returns correct content type and encoding."""
+    filepath = pathlib.Path(__file__).parent / ("hello.txt" + extension)
```

Comment:
This is probably better, than concat.
```suggestion
    filepath = pathlib.Path(__file__).parent / f"hello.txt{extension}"
```

Reply:
Why is that better?

Reply:
String concatenation is generally discouraged. It's slower, for example. A non-nested f-string is a bit easier to parse for humans too.